cmd/kubeadm/app/util/chroot.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"syscall"
+)
+
+// MaybeChroot chroot()s to the new path, if non-empty.
+// NB: All file paths after this call are effectively relative to
+// `rootfs`
+func MaybeChroot(rootfs string) error {
+	if rootfs == "" {
+		return nil
+	}
+
+	if err := syscall.Chroot(rootfs); err != nil {
+		return fmt.Errorf("unable to chroot to %s: %v", rootfs, err)
+	}
```

Comment:
`syscall.Chroot()` does not exist in the `syscall` package for Windows, so this has to be branched away with `runtime.GOOS != "windows"` if native Windows containers would like to run `kubeadm`.

having said that i don't know the state of the Win32 port.

Reply:
yep, file suffixes are cleaner even in the case of no-op function for one of the OSes.

Reply:
I added a `chroot_windows.go` which just returns a more descriptive error when `rootfs != ""`